src/tutorials/boss_tutorial.md

```diff
@@ -0,0 +1,4330 @@
+        Task.Wait(1)
+        target = tmpTarget
+        tmpTarget = nil
+    end
+
+    -- If a player enters the shoot perimeter of the AI, add
+    -- them to the shootTargets table
+    function OnPlayerEnterShoot(trigger, player)
+        if Object.IsValid(player) and player:IsA("Player") then
+            table.insert(shootTargets, player)
+
+            -- Enable the players weapon if they are in the perimeter
+            Events.Broadcast("EnableWeapon", player)
+        end
+    end
+
+    -- If the player leaves the shoot perimeter of the AI, remove
+    -- them from the shootTargets table.
+    function OnPlayerLeftShoot(trigger, player)
+        if Object.IsValid(player) and player:IsA("Player") then
+
+            -- Disable the players weapon if they are out of the perimeter
+            Events.Broadcast("DisableWeapon", player)
+
+            for index, value in ipairs(shootTargets) do
+                if value == player then
+                    table.remove(shootTargets, index)
+
+                    if(target == player and Object.IsValid(BOSS_GEO)) then
+                        BOSS_GEO:StopRotate()
+                    end
+
+                    break
+                end
+            end
+        end
+    end
+
+    -- When the AI receives damage, broadcast that to the player
+    -- who caused the damage.
+    local function OnDamaged(obj, damage)
+        if not sentUpdateBossHealthBar then
+            Events.BroadcastToAllPlayers("CanUpdateBossHealthBar")
+            sentUpdateBossHealthBar = true
+        end
+
+        Events.BroadcastToPlayer(damage.sourcePlayer, "ShowDamage", damage.amount, false)
+    end
+
+    -- Each generator that is disabled, increment the counter so
+    -- that once all 3 are disabled, the AI becomes vulnerable.
+    local function GeneratorDisabled()
+        generatorsDisabled = generatorsDisabled + 1
+
+        if generatorsDisabled == 3 then
+            DAMAGEABLE.isInvulnerable = false
+            Events.BroadcastToAllPlayers("EnableBossHealthBar")
+        end
+    end
+
+    DAMAGEABLE.damagedEvent:Connect(OnDamaged)
+
+    SHOOT_PERIMETER.beginOverlapEvent:Connect(OnPlayerEnterShoot)
+    SHOOT_PERIMETER.endOverlapEvent:Connect(OnPlayerLeftShoot)
+
+    -- Loop through the activities table and add them to
+    -- the activity handler
+    for k, a in pairs(activities) do
+        ACTIVITY_HANDLER:AddActivity(k, activities[k])
+    end
+
+    Events.Connect("GeneratorDisabled", GeneratorDisabled)
+    ```
+
+## Update Boss Health Bar
+
+When the boss is vulnerable to damage, the health bar needs to be updated so that players are aware the boss can receive damage. To do this, the color of the boss name and health bar will be changed when it is vulnerable.
+
+### Update BossHealthUIClient Script
+
+The **BossHealthUIClient** script needs to change the color of the various UI components that build up the heath bar, and boss name.
+
+1. Add **BossName** custom property.
```

Comment:
Might be worth reinforcing custom properties with a picture of each state to check against.